src/components/gui/tabs/query-tab.tsx

```diff
@@ -357,3 +359,33 @@ export default function QueryWindow({
     </ResizablePanelGroup>
   );
 }
+
+function getSingleTableName(query: string): string | null {
+  // Normalize query by removing extra spaces and converting to lowercase
+  const normalizedQuery = query.replace(/\s+/g, " ").trim().toLowerCase();
+
+  // Match the table names after "from" keyword
+  const fromMatch = normalizedQuery.match(/from\s+([^\s,;]+)/i);
+  const joinMatches = normalizedQuery.match(/join\s+([^\s,;]+)/gi);
+
+  // If there are JOINs, more than one table is referenced
+  if (joinMatches && joinMatches.length > 0) {
+    return null;
+  }
+
+  // Check if a single table is present
+  if (fromMatch) {
+    const tableName = fromMatch[1];
+
+    // Ensure no additional tables are mentioned
+    const additionalTablesMatch = normalizedQuery.match(/,\s*[^\s,;]+/);
+    if (additionalTablesMatch) {
+      return null;
+    }
+
+    return tableName;
+  }
+
+  // No table found
+  return null;
+}
```

Comment:
I think it is safer to also build one try and catch and return null if there is unexpected error somewhere. This ensures that it will break anything